src/Parser/RestoDruid/Modules/Traits/Grovewalker.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import calculateEffectiveHealing from 'Parser/Core/calculateEffectiveHealing';
+import SPELLS from 'common/SPELLS';
+
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+import {HEALS_MASTERY_STACK} from '../../Constants';
+
+
+const GROVEWALKER_HEALING_INCREASE = 0.01;
+
+/**
+ * Grovewalker (Artifact Trait)
+ * Increases all healing over time you do by 1%
+ */
+class Grovewalker extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  rank = 0;
+  healing = 0;
+
+  on_initialized() {
+    this.rank = this.combatants.selected.traitsBySpellId[SPELLS.GROVEWALKER.id];
+    this.active = this.rank > 0;
+  }
+
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+    if(!HEALS_MASTERY_STACK.indexOf(spellId)) {
+      return;
+    }
+    if(SPELLS.REGROWTH.id === spellId && !event.tick) {
+      return;
+    }
```

Comment:
We're doing a check that boils down to "is this a resto HoT tick?" a lot. Maybe should be a util function somewhere.

Reply:
This check is only relevant for regrowth as they use the same Id for the HoT part and the heal part.

Reply:
I know why it's needed, I'm just saying we're doing this special case a lot, and it's all in service of detecting if a heal event was a resto HoT tick.